### PR TITLE
Add CLI

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Adds a CLI with two commands (`generate-key`, `register-node`)
- Refactors the config to better handle CLI commands

## Why this way?

I spent a fair bit of time on Friday looking at different "CLI frameworks" and tools and ultimately decided not to use any of them.  [urfave/cli](https://cli.urfave.org/v2/examples/flags/), [Cobra](https://cobra.dev/), and [Kong](https://github.com/alecthomas/kong) all require a bunch of framework-specific boilerplate for little upside.

Even after you've written all that boilerplate, it's still a bunch of work to map those CLI args to a nice config struct like what we have.

The requirements we have for command line parsing are pretty simple:

- Parse the command line arguments and environment variables into a nice struct
- Validate required fields are present (with different required fields per command)
- Show good errors when required fields are missing or invalid values are set
- Generate help text

## Known issues

I haven't figured out how to get good help text for the command-specific arguments. If I put everything into one big command struct, then the required fields for one command become required for every command. That's why I have broken up the `globalOptions` and the command-specific option structs.

Will keep working to make sure we have good help text

### CODE REVIEW

**Comment by neekolas** [comment]
* **#164** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/164?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by richardhuaaa — APPROVED** [review]
